runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/tormenteddemons/TormentedDemonPlugin.java

```diff
@@ -0,0 +1,138 @@
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.GraphicsObject;
+import net.runelite.api.InventoryID;
+import net.runelite.api.Item;
+import net.runelite.api.ItemComposition;
+import net.runelite.api.events.GraphicsObjectCreated;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.events.ConfigChanged;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.plugins.microbot.Microbot;
+
+import java.awt.datatransfer.StringSelection;
+import java.util.StringJoiner;
+import java.util.concurrent.Executors;
+import net.runelite.client.plugins.microbot.util.tile.Rs2Tile;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+import javax.inject.Inject;
+import java.awt.*;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+@PluginDescriptor(
+        name = PluginDescriptor.zerozero + "Tormented Demons",
+        description = "Automates restocking, prayer flicking, and gear switching during Tormented Demon",
+        tags = {"tormented", "flicker", "weapon", "switch"},
+        enabledByDefault = false
+)
+@Slf4j
+public class TormentedDemonPlugin extends Plugin {
+
+    private ScheduledExecutorService scheduledExecutorService;
+
+    @Inject
+    private ConfigManager configManager;
+
+    @Inject
+    private TormentedDemonConfig config;
+
+    @Provides
+    TormentedDemonConfig provideConfig(ConfigManager configManager) {
+        return configManager.getConfig(TormentedDemonConfig.class);
+    }
+
+    @Inject
+    private OverlayManager overlayManager;
+
+    @Inject
+    private TormentedDemonOverlay tormentedDemonOverlay;
+
+    @Inject
+    private TormentedDemonScript tormentedDemonScript;
+
+    @Override
+    protected void startUp() throws AWTException {
+        scheduledExecutorService = Executors.newSingleThreadScheduledExecutor();
+        if (overlayManager != null) {
+            overlayManager.add(tormentedDemonOverlay);
+        }
+        tormentedDemonScript.run(config);
+    }
+
+    @Override
+    protected void shutDown() {
+        tormentedDemonScript.shutdown();
+        overlayManager.remove(tormentedDemonOverlay);
+        if (scheduledExecutorService != null && !scheduledExecutorService.isShutdown()) {
+            scheduledExecutorService.shutdown();
+        }
+    }
+
+    @Subscribe
+    private void onConfigChanged(ConfigChanged event) {
+        if (event.getGroup().equals("tormenteddemon")) {
+            if (event.getKey().equals("fullAuto") && config.fullAuto()) {
+                configManager.setConfiguration("tormenteddemon", "combatOnly", false);
+            } else if (event.getKey().equals("combatOnly") && config.combatOnly()) {
+                configManager.setConfiguration("tormenteddemon", "fullAuto", false);
+            }
+            if (event.getKey().equals("copyGear")) {
```

Comment:
Interesting...what is the copy gear used for?

Reply:
This copies the names your current gear to a comma separated string, so you can add them to the gear switcher quicker/easier

Reply:
oh cool, def something we could use for other plugins aswell